### PR TITLE
fix: leverage gov proposal migrations

### PR DESCRIPTION
**Pull request by robert-zaremba**
## Description

closes: #1627

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: app/upgradev3_3/migrate.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+// Migrates leverage module from v1 to v3
+func (m migrator) migrate(ctx sdk.Context) error {
+	logger := ctx.Logger()
+	if done {
+		logger.Error("Migration already done")
+		return nil
+	}
+	done = true
+
+	// m.gov.RemoveFromActiveProposalQueue(ctx types.Context, proposalID uint64, endTime time.Time)
```
Comment (toteki) [inline app/upgradev3_3/migrate.go:48]: Should we comment this in or remove it?

File: app/upgrades.go [hunk 2]
```diff
@@ -22,34 +23,60 @@ func (app UmeeApp) RegisterUpgradeHandlers(experimental bool) {
...
+			return app.mm.RunMigrations(ctx, app.configurator, fromVM)
+		})
 }
 
-// performs upgrade from v3.1 -> v3.2
-func (app *UmeeApp) registerV3_2Upgrade(_ upgradetypes.Plan) {
-	const planName = "v3.2"
+// performs upgrade from v3.2 -> v3.3
+func (app *UmeeApp) registerV3_2to3_3_Upgrade(_ upgradetypes.Plan) {
+	const planName = "v3.3"
```
Comment (toteki) [inline app/upgrades.go:57]: maybe `"v3.2-v3.3"` for consistency

File: x/leverage/migrations/mv2/migrate.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+func Migrator1to2(gk govkeeper.Keeper) module.MigrationHandler {
+	m := migrator{gk}
+	return m.migrate
+}
+
+var done bool
+
+// Migrates leverage module from v1 to v3
```
[the remark this answers is not in the document]
Comment (robert-zaremba) [reply x/leverage/migrations/mv2/migrate.go]: we don't change consensus version, so yes, should be v3.3

File: app/upgradev3_3/migrate.go (hunk 1 above)
Comment (robert-zaremba) [reply app/upgradev3_3/migrate.go:48]: removed

File: app/upgrades.go (hunk 2 above)
Comment (robert-zaremba) [reply app/upgrades.go:57]: In general for direct migration (`y.x -> y.(x+1)`) we were using `y.(x+1)` name. 
But I don't have strong preference, we can use `"v3.2-v3.3"`

Comment (robert-zaremba) [reply app/upgrades.go]: updated to "v3.2-v3.3"

**Comment by toteki** [comment]
lint: `don't use an underscore in package name` (upgrade3_3)

File: app/upgradev3_3/migrate.go [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+		(*gov.Content)(nil),
+		&UpdateRegistryProposal{},
+	)
+}
+
+type migrator struct {
+	gov govkeeper.Keeper
+}
+
+// Creats migration handler for gov leverage proposals to new the gov  system
```
Comment (toteki) [inline app/upgradev3_3/migrate.go:29]:
```suggestion
// Creats migration handler for gov leverage proposals to new the gov system
```

**Review by toteki — APPROVED** [review]
Once the package name lint is resolved, this looks good. How extensive was our local testing?

**Pull request merged**